sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlCheckBoxGroupFieldForMap.java

```diff
@@ -0,0 +1,205 @@
+	}
+
+	private void uncheckAll() {
+		for (CheckBox checkBox : checkBoxes.values()) {
+			checkBox.setChecked(false);
+		}
+	}
+
+	private void removeAllItems() {
+		checkBoxes.clear();
+		checkBoxesFrame.removeAllViews();
+	}
+
+	@Override
+	protected void initialize(Context context, AttributeSet attrs, int defStyle) {
+		// Nothing to initialize
+	}
+
+	@Override
+	protected void inflateView(Context context, AttributeSet attrs, int defStyle) {
+		LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+
+		if (inflater != null) {
+			inflater.inflate(R.layout.control_checkboxgroup_layout, this);
+		} else {
+			throw new RuntimeException("Unable to inflate layout in " + getClass().getName());
+		}
+	}
+
+	@Override
+	protected void onFinishInflate() {
+		super.onFinishInflate();
+
+		checkBoxesFrame = this.findViewById(R.id.checkboxes_frame);
+	}
+
+	@Override
+	protected Map<T, Boolean> getFieldValue() {
+		Map<T, Boolean> selectedElements = new HashMap<>();
+
+		for (T key : checkBoxes.keySet()) {
+			if (checkBoxes.get(key).isChecked()) {
+				selectedElements.put(key, true);
+			}
+		}
+
+		return selectedElements;
+	}
+
+	@Override
+	protected void setFieldValue(Map<T, Boolean> value) {
+		if (value == null) {
+			uncheckAll();
+		} else {
+			for (Map.Entry<T, Boolean> element : value.entrySet()) {
+				CheckBox checkBox = checkBoxes.get(element.getKey());
+
+				if (checkBox == null) {
+					throw new IllegalArgumentException(
+						"Passed list arguments contains an element that is not part of this ControlCheckBoxGroupField");
+				}
+
+				checkBox.setChecked(element.getValue());
+			}
+		}
+	}
+
+	@Override
+	public void setEnabled(boolean enabled) {
+		// TODO: Implement; Probably replace elements with tags
+	}
+
+	@Override
+	protected void requestFocusForContentView(View nextView) {
+		// Not needed
+	}
+
+	@Override
+	public void setHint(String hint) {
+		// Not needed
+	}
+
+	// Data binding, getters & setters
+	@BindingAdapter("value")
+	public static <E extends Enum<?>> void setValue(ControlCheckBoxGroupFieldForMap<E> view, Map<E, Boolean> value) {
+		view.setFieldValue(value);
+	}
+
+	@InverseBindingAdapter(attribute = "value", event = "valueAttrChanged")
+	public static <E extends Enum<?>> Map<E, Boolean> getValue(ControlCheckBoxGroupFieldForMap<E> view) {
+		return view.getFieldValue();
+	}
+
+	@BindingAdapter("valueAttrChanged")
+	public static <E extends Enum<?>> void setListener(final ControlCheckBoxGroupFieldForMap<E> view, InverseBindingListener listener) {
+		view.inverseBindingListener = listener;
+	}
+
+	@BindingAdapter(value = {
+		"value",
+		"enumClass" })
+	public static <E extends Enum<?>> void setValue(ControlCheckBoxGroupFieldForMap<E> view, Map<E, Boolean> value, Class<E> enumClass) {
+		if (enumClass != null) {
+			view.setEnumClass(enumClass);
+		}
+		view.setFieldValue(value);
+	}
+
+	@Override
+	protected void changeVisualState(VisualState state) {
```

Comment:
please remove the TODO , it's just not needed

Reply:
removed